internal/emigration/emigrate-nobles.lua

```diff
@@ -0,0 +1,352 @@
+    unit.animal.leave_countdown = 2
+
+    -- remove current job
+    if unit.job.current_job then dfhack.job.removeJob(unit.job.current_job) end
+
+    -- break up any social activities
+    for _, actId in ipairs(unit.social_activities) do
+        local act = df.activity_entry.find(actId)
+        if act then act.events[0].flags.dismissed = true end
+    end
+
+    removeUnitFromSiteEntity(unit, histFig, fortEnt)
+    local siteGov = addHistFigToSite(histFig, toSite)
+
+    -- announce the changes
+    local unitName = dfhack.df2console(dfhack.units.getReadableName(unit))
+    local siteName = dfhack.df2console(dfhack.translation.translateName(toSite.name, true))
+    local govName = dfhack.df2console(dfhack.translation.translateName(siteGov.name, true))
+    local line = unitName .. " has left to join " ..govName.. " as lord of " .. siteName .. "."
+    print("[+] "..dfhack.df2console(line))
+    dfhack.gui.showAnnouncement(line, COLOR_WHITE)
+end
+
+---@param unit df.unit
+local function inStrangeMood(unit)
+    local job = unit.job.current_job
+    if not job then return false end
+
+    local jobType = job.job_type -- taken from notifications::for_moody()
+    return df.job_type_class[df.job_type.attrs[jobType].type] == 'StrangeMood'
+end
+
+---@param unit df.unit
+local function isSoldier(unit)
+    return unit.military.squad_id ~= -1
+end
+
+local function listNoblesFound(nobleList)
+    for _, record in pairs(nobleList) do
+        local unit = record.unit
+        local site = record.site
+
+        local nobleName = dfhack.df2console(dfhack.units.getReadableName(unit))
+        local unitMsg = unit.id..": "..nobleName
+        if isSoldier(unit) then
+            local squad = df.squad.find(unit.military.squad_id)
+            if not squad then qerror("could not find unit's squad") end
+            local squadName = dfhack.df2console(dfhack.translation.translateName(squad.name, true))
+            unitMsg = "[!] "..unitMsg.." - soldier in "..squadName
+        else
+            local siteName = dfhack.df2console(dfhack.translation.translateName(site.name, true))
+            unitMsg = unitMsg.." to be sent to "..siteName
+        end
+
+        print(unitMsg)
+    end
+end
+
+local function main()
+    local freeloaders = {}
+    for _, unit in ipairs(dfhack.units.getCitizens()) do
+        if options.unitId ~= -1 and unit.id ~= options.unitId then goto continue end
+        if dfhack.units.isDead(unit) or not dfhack.units.isSane(unit) then goto continue end
+
+        addNobleOfOtherSite(unit, freeloaders)
+        ::continue::
+    end
+
+    if #freeloaders == 0 then
+        if options.unitId == -1 then
+            print("No eligible nobles to be emigrated.")
+        else
+            print("No eligible nobles found with ID = "..options.unitId)
+        end
+    end
+
+    if options.list then
+        listNoblesFound(freeloaders)
+        return
+    end
+
+    for _, record in pairs(freeloaders) do
```

Comment:
if `freeloaders` is a list, then this should be `ipairs` to match the semantics, not `pairs`